e2e benchmark CI is unstable

### Which jobs are failing?

For example, current CI job:
https://github.com/kubernetes-sigs/kwok/actions/runs/16498369654/job/46649813828

similiarly previous CI job:

https://github.com/kubernetes-sigs/kwok/actions/runs/16421011496/job/46399286703


### Which tests are failing?
```
=== RUN   TestBenchmark/Benchmark_Hack/Delete_pods
{"time":"2025-07-24T13:37:49.314425637Z","level":"INFO","source":{"function":"sigs.k8s.io/kwok/pkg/kwokctl/cmd/scale.runE","file":"/home/runner/work/kwok/kwok/pkg/kwokctl/cmd/scale/scale.go","line":112},"msg":"No resource found, use default resource","cluster":"kwok-e2e-benchmark-fdf7f","resource":"pod"}
    benchmark.go:48: Pod 10000
    benchmark.go:48: Pod 9332
    benchmark.go:48: Pod 8593
    benchmark.go:48: Pod 7903
    benchmark.go:48: Pod 7243
    benchmark.go:48: Pod 6652
    benchmark.go:48: Pod 6064
    benchmark.go:48: Pod 5509
    benchmark.go:48: Pod 4976
    benchmark.go:48: Pod 4489
    benchmark.go:48: Pod 4022
    benchmark.go:48: Pod 3592
    benchmark.go:48: Pod 3144
    benchmark.go:48: Pod 2729
    benchmark.go:48: Pod 2325
    benchmark.go:48: Pod 1935
    benchmark.go:48: Pod 1598
    benchmark.go:48: Pod 1283
    benchmark.go:48: Pod 949
    benchmark.go:48: Pod 639
    benchmark.go:48: Pod 382
    benchmark.go:48: Pod 147
{"time":"2025-07-24T13:38:48.841896098Z","level":"INFO","source":{"function":"sigs.k8s.io/kwok/pkg/kwokctl/scale.Scale","file":"/home/runner/work/kwok/kwok/pkg/kwokctl/scale/scale.go","line":219},"msg":"Deleted resources","cluster":"kwok-e2e-benchmark-fdf7f","name":"fake-pod","replicas":0,"resource":"pods","namespace":"default","counter":10000,"elapsed":{"nanosecond":59506929638,"human":"59.506929638s"}}
    benchmark.go:225: failed to wait for resource: context deadline exceeded
--- FAIL: TestBenchmark (157.32s)
    --- FAIL: TestBenchmark/Benchmark_Hack (157.32s)
        --- PASS: TestBenchmark/Benchmark_Hack/Create_nodes (47.23s)
        --- PASS: TestBenchmark/Benchmark_Hack/Create_pods (50.03s)
        --- FAIL: TestBenchmark/Benchmark_Hack/Delete_pods (60.06s)
FAIL
I0724 13:38:49.336056   10448 command.go:38] "Found Provider tooling already installed on the machine" command="/home/runner/work/kwok/kwok/bin/linux/amd64/kwokctl"
I0724 13:38:49.401328   10448 kwok.go:172] Destroying kwok cluster kwok-e2e-benchmark-fdf7f
I0724 13:38:49.401371   10448 command.go:38] "Found Provider tooling already installed on the machine" command="/home/runner/work/kwok/kwok/bin/linux/amd64/kwokctl"
I0724 13:38:49.496036   10448 kwok.go:186] Removing kubeconfig file /tmp/kwok-cluster-kwok-e2e-benchmark-fdf7f-kubecfg202969695
FAIL	sigs.k8s.io/kwok/test/e2e/kwokctl/benchmark	197.967s
FAIL
------------------------------
Test e2e/kwokctl/benchmark failed.
================================================================================
Error: Some tests failed
 - e2e/kwokctl/benchmark
Error: Process completed with exit code 1.
```

```
=== RUN   TestBenchmarkWithHack
=== RUN   TestBenchmarkWithHack/Benchmark_Hack
=== RUN   TestBenchmarkWithHack/Benchmark_Hack/Create_nodes
    benchmark.go:140: Node 0/1 => 5000, 2025-07-24 14:33:10.755660185 +0000 UTC m=+10.207235624
    benchmark.go:140: Node 414/540 => 5000, 2025-07-24 14:33:11.758535557 +0000 UTC m=+11.210110995
    benchmark.go:147: Error Node gap too large, actual: 126, expected: 100, retrying...
    benchmark.go:140: Node 607/904 => 5000, 2025-07-24 14:33:12.759155892 +0000 UTC m=+12.210731340
    benchmark.go:147: Error Node gap too large, actual: 297, expected: 100, retrying...
    benchmark.go:140: Node 735/1285 => 5000, 2025-07-24 14:33:13.767027496 +0000 UTC m=+13.218602944
    benchmark.go:147: Error Node gap too large, actual: 550, expected: 100, retrying...
    benchmark.go:140: Node 869/1624 => 5000, 2025-07-24 14:33:14.767083046 +0000 UTC m=+14.218658495
    benchmark.go:147: Error Node gap too large, actual: 755, expected: 100, retrying...
    benchmark.go:140: Node 1040/1971 => 5000, 2025-07-24 14:33:15.767346801 +0000 UTC m=+15.218922249
    benchmark.go:147: Error Node gap too large, actual: 931, expected: 100, retrying...
    benchmark.go:140: Node 1189/2362 => 5000, 2025-07-24 14:33:16.768603515 +0000 UTC m=+16.220178963
    benchmark.go:147: Error Node gap too large, actual: 1173, expected: 100, retrying...
    benchmark.go:140: Node 1316/2709 => 5000, 2025-07-24 14:33:17.770268691 +0000 UTC m=+17.221844139
    benchmark.go:147: Error Node gap too large, actual: 1393, expected: 100, retrying...
    benchmark.go:140: Node 1462/3029 => 5000, 2025-07-24 14:33:18.770342258 +0000 UTC m=+18.221917706
    benchmark.go:147: Error Node gap too large, actual: 1567, expected: 100, retrying...
    benchmark.go:140: Node 1601/3342 => 5000, 2025-07-24 14:33:19.770856035 +0000 UTC m=+19.222431483
    benchmark.go:147: Error Node gap too large, actual: 1741, expected: 100, retrying...
    benchmark.go:140: Node 1748/3660 => 5000, 2025-07-24 14:33:20.775124893 +0000 UTC m=+20.226700351
    benchmark.go:147: Error Node gap too large, actual: 1912, expected: 100, retrying...
    benchmark.go:140: Node 1867/4005 => 5000, 2025-07-24 14:33:21.776427423 +0000 UTC m=+21.228002871
    benchmark.go:147: Error Node gap too large, actual: 2138, expected: 100, retrying...
    benchmark.go:140: Node 2011/4313 => 5000, 2025-07-24 14:33:22.776869826 +0000 UTC m=+22.228445274
    benchmark.go:147: Error Node gap too large, actual: 2302, expected: 100, retrying...
    benchmark.go:140: Node 2139/4624 => 5000, 2025-07-24 14:33:23.777661403 +0000 UTC m=+23.229236851
    benchmark.go:147: Error Node gap too large, actual: 2485, expected: 100, retrying...
    benchmark.go:140: Node 2265/4923 => 5000, 2025-07-24 14:33:24.777928928 +0000 UTC m=+24.229504376
    benchmark.go:147: Error Node gap too large, actual: 2658, expected: 100, retrying...
    benchmark.go:140: Node 2445/5000 => 5000, 2025-07-24 14:33:25.789298183 +0000 UTC m=+25.240873631
    benchmark.go:147: Error Node gap too large, actual: 2555, expected: 100, retrying...
    benchmark.go:140: Node 2797/5000 => 5000, 2025-07-24 14:33:26.794154656 +0000 UTC m=+26.245730094
    benchmark.go:147: Error Node gap too large, actual: 2203, expected: 100, retrying...
    benchmark.go:140: Node 3186/5000 => 5000, 2025-07-24 14:33:27.795964914 +0000 UTC m=+27.247540361
    benchmark.go:147: Error Node gap too large, actual: 1814, expected: 100, retrying...
    benchmark.go:140: Node 3572/5000 => 5000, 2025-07-24 14:33:28.797406656 +0000 UTC m=+28.248982103
    benchmark.go:147: Error Node gap too large, actual: 1428, expected: 100, retrying...
    benchmark.go:140: Node 3896/5000 => 5000, 2025-07-24 14:33:29.802645268 +0000 UTC m=+29.254220716
    benchmark.go:147: Error Node gap too large, actual: 1104, expected: 100, retrying...
    benchmark.go:140: Node 4183/5000 => 5000, 2025-07-24 14:33:30.802703346 +0000 UTC m=+30.254278794
    benchmark.go:147: Error Node gap too large, actual: 817, expected: 100, retrying...
    benchmark.go:140: Node 4385/5000 => 5000, 2025-07-24 14:33:31.804227274 +0000 UTC m=+31.255802722
    benchmark.go:150: Error Node gap too large, actual: 615, expected: 100
    benchmark_hack.go:174: failed to wait for resource: gap too large for resource Node
--- FAIL: TestBenchmarkWithHack (22.05s)
    --- FAIL: TestBenchmarkWithHack/Benchmark_Hack (22.05s)
        --- FAIL: TestBenchmarkWithHack/Benchmark_Hack/Create_nodes (22.05s)
```

### Since when has it been failing?

from time to time

### Reason for failure (if possible)

60 seconds timeout is not enough for all pods to be deleted

### Anything else we need to know?

_No response_

--- a/test/e2e/benchmark.go
+++ b/test/e2e/benchmark.go
@@ -217,7 +217,7 @@ func CaseBenchmark(kwokctlPath, clusterName string) *features.FeatureBuilder {
 			return ctx
 		}).
 		Assess("Delete pods", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
-			ctx0, cancel := context.WithTimeout(ctx, 120*time.Second)
+			ctx0, cancel := context.WithTimeout(ctx, 90*time.Second)
 			defer cancel()
 
 			err := scaleDeletePod(ctx0, t, kwokctlPath, clusterName, 10000)

--- a/test/e2e/benchmark.go
+++ b/test/e2e/benchmark.go
@@ -217,7 +217,7 @@ func CaseBenchmark(kwokctlPath, clusterName string) *features.FeatureBuilder {
 			return ctx
 		}).
 		Assess("Delete pods", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
-			ctx0, cancel := context.WithTimeout(ctx, 60*time.Second)
+			ctx0, cancel := context.WithTimeout(ctx, 120*time.Second)
 			defer cancel()
 
 			err := scaleDeletePod(ctx0, t, kwokctlPath, clusterName, 10000)

--- a/test/e2e/benchmark_hack.go
+++ b/test/e2e/benchmark_hack.go
@@ -166,7 +166,7 @@ func scaleCreateNodeWithHack(ctx context.Context, t *testing.T, kwokctlPath stri
 func CaseBenchmarkWithHack(kwokctlPath, clusterName string) *features.FeatureBuilder {
 	return features.New("Benchmark Hack").
 		Assess("Create nodes", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
-			ctx0, cancel := context.WithTimeout(ctx, 20*time.Second)
+			ctx0, cancel := context.WithTimeout(ctx, 60*time.Second)
 			defer cancel()
 
 			err := scaleCreateNodeWithHack(ctx0, t, kwokctlPath, clusterName, 5000, 100, 20)
@@ -176,7 +176,7 @@ func CaseBenchmarkWithHack(kwokctlPath, clusterName string) *features.FeatureBui
 			return ctx
 		}).
 		Assess("Create pods", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
-			ctx0, cancel := context.WithTimeout(ctx, 20*time.Second)
+			ctx0, cancel := context.WithTimeout(ctx, 60*time.Second)
 			defer cancel()
 
 			err := scaleCreatePodWithHack(ctx0, t, kwokctlPath, clusterName, 10000, 100, 20)
@@ -186,7 +186,7 @@ func CaseBenchmarkWithHack(kwokctlPath, clusterName string) *features.FeatureBui
 			return ctx
 		}).
 		Assess("Delete pods", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
-			ctx0, cancel := context.WithTimeout(ctx, 20*time.Second)
+			ctx0, cancel := context.WithTimeout(ctx, 120*time.Second)
 			defer cancel()
 
 			err := scaleDeletePodWithHack(ctx0, t, kwokctlPath, clusterName, 10000)

--- a/test/e2e/benchmark_hack.go
+++ b/test/e2e/benchmark_hack.go
@@ -166,7 +166,7 @@ func scaleCreateNodeWithHack(ctx context.Context, t *testing.T, kwokctlPath stri
 func CaseBenchmarkWithHack(kwokctlPath, clusterName string) *features.FeatureBuilder {
 	return features.New("Benchmark Hack").
 		Assess("Create nodes", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
-			ctx0, cancel := context.WithTimeout(ctx, 60*time.Second)
+			ctx0, cancel := context.WithTimeout(ctx, 20*time.Second)
 			defer cancel()
 
 			err := scaleCreateNodeWithHack(ctx0, t, kwokctlPath, clusterName, 5000, 100, 20)
@@ -176,7 +176,7 @@ func CaseBenchmarkWithHack(kwokctlPath, clusterName string) *features.FeatureBui
 			return ctx
 		}).
 		Assess("Create pods", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
-			ctx0, cancel := context.WithTimeout(ctx, 60*time.Second)
+			ctx0, cancel := context.WithTimeout(ctx, 20*time.Second)
 			defer cancel()
 
 			err := scaleCreatePodWithHack(ctx0, t, kwokctlPath, clusterName, 10000, 100, 20)
@@ -186,7 +186,7 @@ func CaseBenchmarkWithHack(kwokctlPath, clusterName string) *features.FeatureBui
 			return ctx
 		}).
 		Assess("Delete pods", func(ctx context.Context, t *testing.T, c *envconf.Config) context.Context {
-			ctx0, cancel := context.WithTimeout(ctx, 120*time.Second)
+			ctx0, cancel := context.WithTimeout(ctx, 30*time.Second)
 			defer cancel()
 
 			err := scaleDeletePodWithHack(ctx0, t, kwokctlPath, clusterName, 10000)

